src/go/rpk/pkg/config/params.go

```diff
@@ -234,8 +236,20 @@ func ParamsFromCommand(cmd *cobra.Command) *Params {
 //  * Sets unset default values.
 //
 func (p *Params) Load(fs afero.Fs) (*Config, error) {
+	cf := "/etc/redpanda/redpanda.yaml"
+	// If we have a config path loaded (through --config flag) the user
+	// expect to load or create the file from this directory.
+	if p.ConfigPath != "" {
+		if exist, _ := afero.Exists(fs, p.ConfigPath); !exist {
+			err := fs.MkdirAll(filepath.Dir(p.ConfigPath), 0o755)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
I think we agreed in leaving it since this is what the manager was doing. I'm fine removing it in a different commit in this PR to see which CI test fails.

PR note:
The old manager created a directory when flag --config was passed, we will keep this behavior for now.

Reply:
Ok we'll leave it in this PR but I think we should remove it during this release in a followup PR. It's pretty surprising behavior to me